cli/compose/convert/volume.go

```diff
@@ -61,17 +61,18 @@ func convertVolumeToMount(
 		return result, errors.Errorf("undefined volume %q", volume.Source)
 	}
 
+	if stackVolume.External.External && stackVolume.External.Name != "" {
+		return result, errors.Errorf("cannot specify external volume name %q", stackVolume.External.Name)
+	}
 	result.Source = namespace.Scope(volume.Source)
 	result.VolumeOptions = &mount.VolumeOptions{}
 
 	if volume.Volume != nil {
 		result.VolumeOptions.NoCopy = volume.Volume.NoCopy
 	}
 
-	// External named volumes
-	if stackVolume.External.External {
-		result.Source = stackVolume.External.Name
-		return result, nil
```

Comment:
I think the warning would go around here: https://github.com/docker/cli/blob/732261f774f724a44f79c3138459cf190cb01558/cli/compose/loader/loader.go#L447

Probably as an `else` to that if block

I think the `validate` problem was fixed on master, so if you rebase it should be fixed.